### PR TITLE
[zh] Sync #13899 into Chinese

### PR DESCRIPTION
**Pull request by wilsonwu**
Sync #13899 into Chinese with some format improvement:
```
content/zh/boilerplates/gateway-api-gamma-support.md
content/zh/docs/tasks/traffic-management/ingress/ingress-sni-passthrough/index.md
content/zh/docs/tasks/traffic-management/tcp-traffic-shifting/index.md
```

- [ ] Ambient
- [x] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Extensions and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: content/zh/boilerplates/gateway-api-gamma-support.md [hunk 1]
```diff
@@ -1,8 +1,26 @@
...
+{{< boilerplate gateway-api-future >}}
+{{< boilerplate gateway-api-choose >}}
+{{< /tip >}}
 
 {{< warning >}}
 请注意，本文使用 Gateway API 配置内部网格（东西）流量，即不仅是 Ingress（南北）流量。
 使用 Gateway API 配置内部网格流量目前是一个还在开发的[实验性特性](https://gateway-api.sigs.k8s.io/geps/overview/#status)。
+If using the Gateway API instructions, before proceeding make sure to:
+
+1) 安装 Gateway API CRD 的 **实验版本** 版本：
```
Comment (windsonsea) [inline content/zh/boilerplates/gateway-api-gamma-support.md:13]:
```suggestion
1) 安装 **实验版本** 的 Gateway API CRD：
```

File: content/zh/boilerplates/gateway-api-gamma-support.md [hunk 2]
```diff
@@ -1,8 +1,26 @@
...
-{{< boilerplate gateway-api-support >}}
+{{< tip >}}
+{{< boilerplate gateway-api-future >}}
+{{< boilerplate gateway-api-choose >}}
+{{< /tip >}}
 
 {{< warning >}}
 请注意，本文使用 Gateway API 配置内部网格（东西）流量，即不仅是 Ingress（南北）流量。
 使用 Gateway API 配置内部网格流量目前是一个还在开发的[实验性特性](https://gateway-api.sigs.k8s.io/geps/overview/#status)。
+If using the Gateway API instructions, before proceeding make sure to:
```
Comment (windsonsea) [inline content/zh/boilerplates/gateway-api-gamma-support.md:11]:
```suggestion
若要使用 Gateway API 指令，请先确保：
```

File: content/zh/boilerplates/gateway-api-gamma-support.md [hunk 3]
```diff
@@ -1,8 +1,26 @@
...
+
+1) 安装 Gateway API CRD 的 **实验版本** 版本：
+
+    {{< text syntax=bash snip_id=install_experimental_crds >}}
+    $ kubectl kustomize "github.com/kubernetes-sigs/gateway-api/config/crd/experimental?ref={{< k8s_gateway_api_version >}}" | kubectl apply -f -
+    {{< /text >}}
+
+2) 安装 Istio 时，通过将 `PILOT_ENABLE_ALPHA_GATEWAY_API`
+    环境变量设置为 `true` 使 Istio 读取 Alpha 版本的资源：
```
Comment (windsonsea) [inline content/zh/boilerplates/gateway-api-gamma-support.md:20]:
```suggestion
    环境变量设置为 `true` 使 Istio 读取 Alpha 版本的 Gateway API 资源：
```

**Comment by ericvn** [comment]
It would be best to split this PR into two, one for the 1.17.6 docs so that can be cherry-picked and the remainder which should NOT be cherry-picked.

**Comment by wilsonwu** [comment]
> It would be best to split this PR into two, one for the 1.17.6 docs so that can be cherry-picked and the remainder which should NOT be cherry-picked (at least I don't see that we are cherry-picking #13899).

Done.

**Review by windsonsea — APPROVED** [review]
Thanks. LGTM

**Pull request merged**